packages/grid/x-data-grid/src/hooks/features/export/useGridPrintExport.tsx

```diff
@@ -272,7 +273,15 @@ export const useGridPrintExport = (
 
       if (props.pagination) {
         const visibleRowCount = gridExpandedRowCountSelector(apiRef);
-        apiRef.current.setPageSize(visibleRowCount);
+        const paginationModel = {
+          page: 0,
+          pageSize: visibleRowCount,
+        };
+        apiRef.current.updateControlState(
+          'pagination',
+          mergeStateWithPaginationModel(visibleRowCount, 'DataGridPro', paginationModel),
```

Comment:
Would be great to leave a comment here to justify the usage of the `DataGridPro` signature.